stan/math/rev/mat/fun/softmax.hpp

```diff
@@ -4,81 +4,78 @@
 #include <stan/math/prim/arr/err/check_nonzero_size.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <stan/math/prim/mat/fun/softmax.hpp>
-#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/mat.hpp>
 #include <vector>
 
 namespace stan {
 namespace math {
 
 namespace {
-class softmax_elt_vari : public vari {
- private:
-  vari** alpha_;
-  const double* softmax_alpha_;
-  const int size_;  // array sizes
-  const int idx_;   // in in softmax output
-
+class softmax_op {
+  int N;
+  double* y_mem_;  // Holds the results of the softmax
```

Comment:
Same principle.  How about `value_` or `values_` or `result_` or even `softmax_value_` if you want to be really clear.

We often take the shortcuts of using `x` for inputs and `y` for outputs.